external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSecurityHelper.scala

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.kafka010
+
+import org.apache.hadoop.security.UserGroupInformation
+import org.apache.hadoop.security.token.{Token, TokenIdentifier}
+import org.apache.kafka.common.security.scram.ScramLoginModule
+
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.Logging
+import org.apache.spark.internal.config._
+
+private[kafka010] object KafkaSecurityHelper extends Logging {
+  def getKeytabJaasParams(sparkConf: SparkConf): Option[String] = {
+    val keytab = sparkConf.get(KEYTAB)
+    if (keytab.isDefined) {
+      val serviceName = sparkConf.get(KAFKA_KERBEROS_SERVICE_NAME)
+      require(serviceName.nonEmpty, "Kerberos service name must be defined")
+      val principal = sparkConf.get(PRINCIPAL)
+      require(principal.nonEmpty, "Principal must be defined")
+
+      val params =
+        s"""
+        |${getKrb5LoginModuleName} required
+        | useKeyTab=true
+        | serviceName="${serviceName.get}"
+        | keyTab="${keytab.get}"
+        | principal="${principal.get}";
+        """.stripMargin.replace("\n", "")
+      logDebug(s"Krb JAAS params: $params")
+      Some(params)
+    } else {
+      None
+    }
+  }
+
+  private def getKrb5LoginModuleName(): String = {
```

Comment:
+ add a comment pointing at hadoop UserGroupInformation so that if someone ever needs to maintain it, they'll know where to look

Reply:
Added.